pkg/protocol/engine/mempool/v1/state_metadata.go

```diff
@@ -69,7 +69,7 @@ func (s *StateMetadata) State() mempool.State {
 	return s.state
 }
 
-func (s *StateMetadata) ConflictIDs() *lpromise.Set[iotago.TransactionID] {
+func (s *StateMetadata) ConflictIDs() reactive.Set[iotago.TransactionID] {
```

Comment:
`reactive.DerivedSet`?

Reply:
```suggestion
func (s *StateMetadata) ConflictIDs() reactive.DerivedSet[iotago.TransactionID] {
```
I agree, that's confusing. It should be the derived set, right?

Reply:
Changed to ReadableSet, so that callers can't modify the contents of the set.

Reply:
Ok lets change that in another PR where we cleanup all sets to use ReadableSets where necessary.